### ISSUE TITLE
Adding alt text for images in documentation

### ISSUE THREAD

**Issue by spcanelon** [quarto-dev/quarto-web#20]
Hello, and thanks for your work on this! It looks like a **great** system and I'm excited to try it out.

I noticed the documentation includes instructions on [how to add alternative (alt) text](https://quarto.org/docs/authoring/figures-and-layout.html#alt-text) within Quarto -- thanks for that! 🙌🏽 

I also noticed there are some images within the documentation itself that don't have alt text, which makes the information they are meant to convey inaccessible to R users that use screen readers. Images without alt text are often read aloud by default as "image" which you can imagine isn't helpful when trying to learn how to use a package. I'm opening this issue to share some encouragement and some resources that I'm hoping can help 🙂 

I haven't checked the images on _all_ of the documentation pages, but here are a few examples of images that still need alt text:
- Figure 1 in [Welcome to Quarto](https://quarto.org/#hello-quarto)
- Data viz in [HTML Code Blocks: Folding Code](https://quarto.org/docs/output-formats/html-code.html#folding-code)
- Screenshots in [HTML Code Blocks: Code Tools](https://quarto.org/docs/output-formats/html-code.html#code-tools), [HTML Code Blocks: Appearance](https://quarto.org/docs/output-formats/html-code.html#appearance), and [HTML Code Blocks: Highlighting](https://quarto.org/docs/output-formats/html-code.html#highlighting)

To determine what kind of images need alt text and how to write alt text I recommend the following resources from W3 and WebAIM:
- [Images Concepts • Images • WAI Web Accessibility Tutorials](https://www.w3.org/WAI/tutorials/images/)
- [An alt Decision Tree • Images • WAI Web Accessibility Tutorials](https://www.w3.org/WAI/tutorials/images/decision-tree/)
- [WebAIM: Alternative Text](https://webaim.org/techniques/alttext/)

Specific to writing alt text for data visualizations:
- [Writing Alt Text for Data Visualization](https://medium.com/nightingale/writing-alt-text-for-data-visualization-2a218ef43f81) by Amy Cesal
- [Revealing Room for Improvement in Accessibility within a Social Media
Data Visualization Learning Community: What makes good alt text?](https://spcanelon.github.io/csvConf2021/slides/#31) by Liz Hare and myself

Examples of R package documentation with alt text added by the package author:
- {stacks}: [Release stacks 0.2.1 • tidymodels/stacks](https://github.com/tidymodels/stacks/releases/tag/v0.2.1) by Simon Couch
- {pointblank}: [Intro to Information Management • pointblank](https://rich-iannone.github.io/pointblank/articles/INFO-1.html) and [Advanced Information Management • pointblank](https://rich-iannone.github.io/pointblank/articles/INFO-2.html#ensuring-that-snippets-and-other-table-metadata-element-are-up-to-date-1) by Rich Iannone

Some resources on the experiences of screen reader users:
- [WebAIM: Screen Reader User Survey #9 Results](https://webaim.org/projects/screenreadersurvey9/)
- [Accessible Data Science Beyond Visual Models: Non-Visual Interactions with R and RStudio Packages](https://www.rstudio.com/resources/rstudioglobal-2021/accessible-data-science-beyond-visual-models-non-visual-interactions-with-r-and-rstudio-packages/) - RStudio Global 2021 talk by JooYoung Seo

Thank you!

**Comment by pommevilla** [comment]
Thanks for these excellent resources @spcanelon! I'll definitely be using them as I go through and add alt-text to these (and other) figures in the docs. Thanks for reporting!

**Comment by spcanelon** [comment]
Yay, thanks so much @pommevilla! I appreciate your willingness to take action. And I'm glad the resources are helpful! Feel free to let me know if I can help find anything more specific -- I'm learning about web accessibility myself so it's possible I'll have some article bookmarked somewhere 🙂

**Comment by pommevilla** [comment]
Hi Silvia, I just pushed a first draft of alt texts for all the images in #23. I learned a lot from those sources you linked. I was also blown away the demonstration JooYoung gave showing how he worked with R. It definitely motivated me to make better alt text and to think about accessibility more 😅 I'll see about getting a preview site up.

**Comment by spcanelon** [comment]
@pommevilla thanks for sharing that, that's really great to hear 🙂
And of course **thank you** for adding the alt text and making the documentation that much more accessible! ⭐

**Comment by jjallaire** [comment]
This is now merged and deployed on https://quarto.org. @spcanelon thanks again!

**Comment by spcanelon** [comment]
Thanks so much @pommevilla and @jjallaire! I really appreciate how quickly and thoughtfully this issue was resolved. 

I know it must have taken considerable time and effort to learn how to write alt text, track down any images missing alt text, and then make all of the changes (not to mention [discover issues along the way](https://github.com/quarto-dev/quarto-web/pull/23#issuecomment-899815600)!). I've been in similar situations and am trying to make it a habit in my own workflow to add the alt text right away when I insert an image so that I don't ever have to do it retroactively! 😅

Knowing that this effort has been put in makes me feel more confident about recommending `quarto` to other R users, so thanks for making it more accessible! I think it could also serve as a **great** example that other package/tool developers can use as a reference when they write their own documentation, particularly because the site happens to include several different types of images 🙂

**Comment by jjallaire** [comment]
I managed to get `fig-alt` onto all of the images within tables here: https://github.com/quarto-dev/quarto-web/commit/3901dc269a39ba8a3a90cdf1af27105f8d515aa2

Had to do two things:

1) Use the visual markdown editor to apply the attributes (so it would auto-format the table to enclose the link correctly)

2) In one case, use a `layout` attribute rather than a table (b/c the length of the link created by `fig-alt` was skewing the sizes of the table columns): https://github.com/quarto-dev/quarto-web/blob/main/docs/websites/website-navigation.qmd#L134-L152

For side by side content I think the `layout` attribute is a much better syntax anyway, and it will always interact correctly with `fig-alt`.
